include/fmt/ranges.h

```diff
@@ -591,10 +597,33 @@ struct formatter<join_view<It, Sentinel, Char>, Char> {
     return value_formatter_.parse(ctx);
   }
 
+#ifdef FMT_USE_ITERATOR_CONCEPT
+  static constexpr bool is_input_iterator =
+      std::input_iterator<It> && !std::forward_iterator<It>;
+#endif
+
   template <typename FormatContext>
+#ifdef FMT_USE_ITERATOR_CONCEPT
+    requires(!is_input_iterator)
+#endif
   auto format(const join_view<It, Sentinel, Char>& value,
               FormatContext& ctx) const -> decltype(ctx.out()) {
     auto it = value.begin;
+    return format_impl(value, ctx, it);
+  }
+
+#ifdef FMT_USE_ITERATOR_CONCEPT
+  template <typename FormatContext>
+    requires is_input_iterator
+  auto format(join_view<It, Sentinel, Char>& value,
+              FormatContext& ctx) const -> decltype(ctx.out()) {
+    return format_impl(value, ctx, value.begin);
+  }
+#endif
```

Comment:
I think we should (1) check if the iterator is copyable rather than trying to detect if it is an input iterator and (2) use SFINAE instead of concepts.

Reply:
Nice suggestion, certainly simpler, have made the change

Other changes introduced are just formatting from running clang-format (17.0.5 as you suggested)